docs/syntax-attributes-functions-parameters.md

```diff
@@ -0,0 +1,61 @@
+# Function Attribute Parameters
+
+## Summary
+
+This RFC proposes a syntax for function attribute parameters. This is a follow up to the [Attributes](./syntax-attributes-functions.md) RFC, which proposed a design for function attributes. We only propose a syntactic extension in this RFC; the rest of the implementation details remain unchanged.
+
+## Motivation
+
+The [Attributes](./syntax-attributes-functions.md) RFC provides a syntax for parameterless function attributes. This suffices for some potential use cases, such as `@native`, and `@inline`. However, other potential cases such as `@deprecated` will benefit from an expressive attribute syntax that allows parameters to be supplied. We can use the parameters to provide the name of the function to be used in place of the deprecated function, and the reason for deprecation. This information can be used to generate informative deprecation warning messages. Another potential use case could be an `@unroll` attribute for loop unrolling which could use a numeric parameter for specifying the number of iterations to be unrolled. This would also require supporting attributes on loops, which would be a topic of discussion for a separate RFC. It might be desirable to allow these attributes to assume a default value for parameters not provided explicitly.
+
+## Design
+
+The following syntax is proposed for attributes:
+
+```ebnf
+parameter-table = '{' '}'
+          | '{' parameter (sep parameter)* '}'
+
+sep = ',' | ';'
+
+parameter = literal 
+          | NAME '=' literal
+
+literal = BOOLEAN | NUMBER | STRING | NIL | parameter-table
+
+list-attribute = NAME parameter-table
+               | NAME
+
+attribute = '@' NAME 
+          | '@[' list-attribute (',' list-attribute)* ']'
+```
+
+In Luau scripts, attributes are specified before the `function` keyword in function definitions. In declaration files, attributes are specified in function type declarations, before the `function` keyword in `declare function (type, ...) : type` syntax, and before the `(` in the `name: (type, ...) : type` syntax. The primary extension proposed to the [Attributes](./syntax-attributes-functions.md) RFC is a new delimited syntax `@[]` for specifying multiple comma-separated attributes with parameters supplied through an optional Luau table. The important features are:
```

Comment:
We discussed this internally, so this part of the RFC needs changing/removed.

```
declare @checked function f(): ()
local @native function f() end
```
is just weird. Let's put them before statements, not interpolated in them.

Reply:
It also violates the original attributes RFC which puts attributes before `local`.

> Multiple attributes are supported inline, so that a function with multiple attributes does not have several lines of boilerplate:
```lua
@attribute1 @attribute2 @attribute3 @attribute4 @attribute5
local function example()
end
```

Reply:
Addressed this and updated the `ebnf` syntax description accordingly.